spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2318,6 +2330,13 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde {
             return None
         }
 
+        for (key <- join.leftKeys) {
+          if (!supportedSortMergeJoinEqualType(key.dataType)) {
+            withInfo(op, s"Unsupported join key type ${key.dataType} on key: ${key.sql}")
+            return None
```

Comment:
nit: do we want to stop checking as soon as we hit one unsupported key type or should we call `withInfo` for each unsupported key type (if there are multiple)

Reply:
Changed it to for each unsupported key type.